### PR TITLE
Provider Soap2Day

### PR DESCRIPTION
**Pull request by Stormunblessed**

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/SoaptwoDayProvider.kt [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                this.name,
+                TvType.TvSeries,
+                image,
+                null,
+                null
+            )
+        }
+    }
+
+    override suspend fun load(url: String): LoadResponse? {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/SoaptwoDayProvider.kt:69]: Not needed, but Similar Movies exist

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/SoaptwoDayProvider.kt [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        val test = try {
+            doc.selectFirst(".col-md-5 > div:nth-child(1) > div:nth-child(1) > img").attr("src") ?: ""
+        } catch (e: Exception) {
+            ""
+        }
+        val ajaxlink = if (test.contains("movie")) "$mainUrl/home/index/GetMInfoAjax" else "$mainUrl/home/index/GetEInfoAjax"
+        listOf(
+            idplayer,
+            idplayer2,
+        ).mapNotNull { playerID ->
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/SoaptwoDayProvider.kt:161]: I think that if you want to do mapNotNull then do ?.text() on line 149 and 150. Also I think you can do filterNotNull().apmap for faster loading

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/SoaptwoDayProvider.kt [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            json.subs?.forEach { subtitle ->
+                val sublink = mainUrl+subtitle.path
+                listOf(
+                    sublink,
+                    subtitle.downlink
+                ).mapNotNull { subs ->
+                    if (subs != null) {
+                        if (subs.isNotBlank()) {
+                            subtitleCallback(
+                                SubtitleFile(subtitle.name!!, subs)
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/SoaptwoDayProvider.kt:209]: if subtitle.name is required then make it not nullable in the Subs class

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/SoaptwoDayProvider.kt [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                }
+            }
+            json.subs?.forEach { subtitle ->
+                val sublink = mainUrl+subtitle.path
+                listOf(
+                    sublink,
+                    subtitle.downlink
+                ).mapNotNull { subs ->
+                    if (subs != null) {
+                        if (subs.isNotBlank()) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/SoaptwoDayProvider.kt:207]: I would do !isBlankOrNull

**Review by LagradOst — APPROVED** [review]
Very good 👍 
Some minor changes are optional, ready to merge when you want.

**Comment by Stormunblessed** [comment]
this it's what my laziness let me do, I think it's ready for merge

**Pull request merged**